### PR TITLE
refactor(wallet): rewrite channel transaction lifecycle 

### PR DESCRIPTION
**Pull request by lerencao**

### CODE REVIEW

File: sgwallet/src/wallet.rs [hunk 1]
```diff
@@ -167,65 +190,93 @@ where
...
+        )?;
+
+        Ok(ChannelTransactionPayload::new_with_write_set(
+            write_set_body,
+            channel_txn_sigs.public_key.clone(),
+            channel_txn_sigs.write_set_payload_signature.clone(),
+        ))
     }
 
     /// Verify channel participant's txn
```
Comment (jolestar) [inline sgwallet/src/wallet.rs]: rename this method to verify_request?

File: sgtypes/src/channel_transaction.rs [hunk 2]
```diff
@@ -1,28 +1,196 @@
...
+    channel_sequence_number: u64,
+    /// The txn expiration time
+    expiration_time: Duration,
+
+    args: Vec<TransactionArgument>,
+
+    /// Maximal total gas specified by wallet to spend for this transaction.
+    max_gas_amount: u64,
+    /// Maximal price can be paid per gas.
+    gas_unit_price: u64,
```
Comment (jolestar) [inline sgtypes/src/channel_transaction.rs]: max_gas_amount, gas_unit_price can be removed from ChannelTransaction. let The sender decide the gas value when submitting the transaction to the chain. 
The expiration_time can keep for verifying ChannelTransaction expiration but not is the RawTransaction 's expiration_time.

File: sgwallet/src/wallet.rs [hunk 3]
```diff
@@ -447,20 +453,121 @@ where
...
+            channel.apply_output(output)?;
+            gas
+        };
+
+        info!("success apply channel request: {}", request_id);
+        Ok(gas)
+    }
+
+    /// called by sender, to verify receiver's response
+    pub fn verify_response(
```
Comment (jolestar) [inline sgwallet/src/wallet.rs]: Should it be a private function?

File: sgtypes/src/channel_transaction.rs (hunk 2 above)
Comment (lerencao) [reply sgtypes/src/channel_transaction.rs]: Updated. gas related fileds are removed from channel transaction.

expiration_time is keeped for later usage.

**Pull request merged**